core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -202,22 +194,19 @@ abstract class AbstractIndex[K, V](@volatile var file: File, val baseOffset: Lon
    */
   def truncateTo(offset: Long): Unit
 
+  protected def safeForceUnmap(): Unit = {
+    try forceUnmap()
+    catch {
+      case t: Throwable => error(s"Error unmapping index $file", t)
+    }
+  }
+
   /**
    * Forcefully free the buffer's mmap.
    */
-  protected def forceUnmap(m: MappedByteBuffer) {
-    try {
-      m match {
-        case buffer: DirectBuffer =>
-          val bufferCleaner = buffer.cleaner()
-          /* cleaner can be null if the mapped region has size 0 */
-          if (bufferCleaner != null)
-            bufferCleaner.clean()
-        case _ =>
-      }
-    } catch {
-      case t: Throwable => error("Error when freeing index buffer", t)
-    }
+  protected[log] def forceUnmap() {
```

Comment:
When would we need `forceUnmap` instead of `safeForceUnmap`?

Reply:
Whenever you want the failure to propagated, which so far is only a test.